pageserver/src/storage_sync2/index.rs

```diff
@@ -91,6 +93,97 @@ pub struct IndexPart {
     metadata_bytes: Vec<u8>,
 }
 
+pub type IndexPart = IndexPartImpl<LayerFileName>;
```

Comment:
This reminds me about our crate that we dont use as much as it deserves :) https://github.com/neondatabase/aversion

It looks like it is sort of another way to evolve format. We have evolved it from v0 to add layer metadata and used an Option to be able to parse both variants.

I'm not sure if it worth investing into it right now, so lets have a comment at least with a reference to the issue so we're able to determine when we can remove it

Reply:
https://github.com/neondatabase/neon/issues/3072